### DISCUSSION TITLE
Isolating logs for an individual container?

### DISCUSSION THREAD

**Discussion by theogravity** [testcontainers/testcontainers-node#270] [Q&A] [answered]
I'm working with a tool that spins up a lot of containers from various services using this library. Is there a way to only get log output from a specific container vs all of them?

**Comment by cristianrgreco** [comment, marked as answer]
Hi @theogravity, you cannot get logs from a single container using the `DEBUG` environment variable as the container IDs and names are random, but you can get the logs of a container at runtime by running:

```javascript
const container = await new GenericContainer("alpine")
  .start();

const stream = await container.logs();
stream
    .on("data", line => console.log(line))
    .on("err", line => console.error(line))
    .on("end", () => console.log("Stream closed"));
```
